File: pinyougou/pyg-sellergoods/pyg-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import com.pinyougou.service.BaseService;
import com.pinyougou.vo.PageResult;

import java.util.List;

public interface BrandService extends BaseService<TbBrand> {
    /**
     * 查询所有品牌信息
     * @return
     */
    List<TbBrand> queryAll();

    /**
     * 测试分页助手和通用mapper
     * @param page
     * @param rows
     * @return
     */
    List<TbBrand> testPage(Integer page, Integer rows);

    PageResult search(Integer page, Integer rows, TbBrand brand);
}

File: pinyougou/pyg-manager-web/src/main/webapp/js/service/brandService.js
//服务层
app.service("brandService",function ($http) {
    //条件分页查询
    this.search = function(page,rows,searchEntity) {
        return $http.post("../brand/search.do?page=" + page + "&rows=" + rows, searchEntity);
    }

    //新增
    this.add = function (entity) {
        return $http.post("../brand/add.do",entity);
    }

    //更新
    this.update = function (entity) {
        return $http.post("../brand/update.do",entity);
    }

    //根据主键查询
    this.findOne = function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    }

    //批量删除
    this.delete = function (ids) {
        return $http.get("../brand/delete.do?ids="+ids);

    }
    //查询品牌列表
    this.selectOptionList = function () {
        return $http.get("../brand/selectOptionList.do");
    };
});